packages/ckeditor5-link/src/linkediting.js

```diff
@@ -414,4 +418,86 @@ export default class LinkEditing extends Plugin {
 			}
 		} );
 	}
+
+	/**
+	 * Starts listening to {@link module:engine/model/model~Model#deleteContent} and {@link module:engine/model/model~Model#insertContent}
+	 * and checks whether typing over the link. If so, attributes of removed text are preserved and applied to the inserted text.
+	 *
+	 * The purpose of this action is to allow modifying a text without loosing the `linkHref` attribute (and other).
+	 *
+	 * See https://github.com/ckeditor/ckeditor5/issues/4762.
+	 *
+	 * @private
+	 */
+	_enableTypingOverLink() {
+		const editor = this.editor;
+
+		// Selection attributes when started typing over the link.
+		let selectionAttributes;
+
+		// Whether pressed `Backspace` or `Delete`. If so, attributes should not be preserved.
+		let deletedContent;
+
+		// Detect pressing `Backspace` / `Delete`.
+		editor.editing.view.document.on( 'delete', ( evt, data ) => {
+			/* istanbul ignore else */
+			if ( data.domEvent.keyCode == keyCodes.delete || data.domEvent.keyCode == keyCodes.backspace ) {
+				deletedContent = true;
+			}
+		}, { priority: 'high' } );
+
+		editor.model.on( 'deleteContent', () => {
+			const selection = editor.model.document.selection;
+
+			// Copy attributes only if anything is selected.
+			if ( selection.isCollapsed ) {
+				return;
+			}
+
+			// When the content was deleted, do not preserve attributes.
+			if ( deletedContent ) {
+				deletedContent = false;
+
+				return;
+			}
+
+			const nodeAfterFirstPosition = selection.getFirstPosition().nodeAfter;
+			const nodeBeforeLastPosition = selection.getLastPosition().nodeBefore;
+
+			if ( isCorrectLink( nodeAfterFirstPosition, nodeBeforeLastPosition ) ) {
+				selectionAttributes = nodeAfterFirstPosition.getAttributes();
+			}
+		}, { priority: 'high' } );
+
+		editor.model.on( 'insertContent', ( evt, [ element ] ) => {
+			if ( !selectionAttributes ) {
+				return;
+			}
+
+			editor.model.change( writer => {
+				for ( const [ attribute, value ] of selectionAttributes ) {
+					writer.setAttribute( attribute, value, element );
+				}
+			} );
+
+			selectionAttributes = null;
+		}, { priority: 'high' } );
+	}
+}
+
+// Checks whether passed elements are the same and they represent a text node.
+//
+// @param {module:engine/model/node~Node} nodeA
+// @param {module:engine/model/node~Node} nodeB
+// @returns {Boolean}
+function isCorrectLink( nodeA, nodeB ) {
```

Comment:
Looking at the function alone, it should have a different name:

*   Does not check the link at all.
*   Checks only it is a single text node.

In a broader scope - This check should be improved, the first case that fails is:

![](https://user-images.githubusercontent.com/247363/87147340-c3997900-c2ac-11ea-969e-70d026f07fb1.gif)

The other thing is that this also works for other attributes and if IIRC, it shouldn't (I might skip something when noting). It should preserve a link with other attributes, so bolded link, but not bold alone:

![Peek 2020-07-10 12-59](https://user-images.githubusercontent.com/247363/87147707-5e925300-c2ad-11ea-89cf-c7373eb547bf.gif)

So those two are tied together in this check. We already have [`findLinkRange()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_link_findlinkrange.html#static-function-findLinkRange) - so it might be used here by comparing range from selection and the result of `findLinkRange()`. This should cover cases like this:

![Selection_420](https://user-images.githubusercontent.com/247363/87148151-29d2cb80-c2ae-11ea-995d-f8859c456528.png)

What to preserve in such case? I think that anything repeatable is good enough. The goal is to preserve link + other attributes in cases when all those attributes are continous (e.g. bolded link). In a case above whatever it will take - I'm guessing the selection attributes will be preserved, so it would end up as a bolded link and it is OK IMO.